[RFC] Fix jailhouse build

mbolivar opened the pull request:
The jailhouse board's build is currently broken.

It wants to depend on the qemu board's devicetree, but can no longer
do so since the board-level dts files were moved into individual board
directories.

To restore the build, dts/x86/ia32_qemu.dtsi, and include it from
qemu_x86.dts as well as x86_jailhouse.dts.

mbolivar commented:
@glima FYI

andrewboie commented:
CI should have caught the breakage, is jailhouse target not compiled by sanitycheck? We need to do that.

nashif commented:
> CI should have caught the breakage, is jailhouse target not compiled by sanitycheck? We need to do that.

We missed the fact the "board" did not add a board.yaml file, thats why it was not caught by CI

nashif commented:
@mbolivar Why make this dependent on qemu dts, I would instead just add the dts file directly into the jailhouse board and avoid this, we might get conflicting changes in the future for both board definitions.

Commit:
arch: x86: fix jailhouse build

The jailhouse board's build is currently broken.

It wants to depend on the qemu board's devicetree, but can no longer
do so since the board-level dts files were moved into individual board
directories.

To restore the build, insert the qemu_x86.dts contents into
x86_jailhouse.dts.

Signed-off-by: Marti Bolivar <marti@opensourcefoundries.com>

mbolivar commented:
@nashif sure, I don't understand this board well which is why I made this an RFC. Now updated.
